packages/rpc-core/src/index.ts

```diff
@@ -250,6 +250,10 @@ export default class Rpc implements RpcInterface {
             );
         };
       }).pipe(
+        // Duplicated in api-derive/util/drr
+        distinctUntilChanged((a: any, b: any): boolean =>
```

Comment:
If i'm not mistaken, `a` and `b` can either be `Codec | Codec[] | null` , and nothing else. I don't think we need to do JSON.stringify, we can just use `.eq` for Codec, `map(.eq)` for Codec[] or `===` for null.

(in api-derive we do return some more complex stuff, so JSON.stringify might be the easiest solution)

Reply:
Yes, started that way. However, it is just overkill since have have proper toJSON

https://github.com/polkadot-js/api/commit/45dc8e766f076a7123af5745dd811f43323c8f9c